pkg/oc/cli/debug/debug.go

```diff
@@ -725,6 +740,66 @@ func containerNames(pod *corev1.Pod) []string {
 
 func (o *DebugOptions) approximatePodTemplateForObject(object runtime.Object) (*corev1.PodTemplateSpec, error) {
 	switch t := object.(type) {
+	case *corev1.Node:
+		if len(o.NodeName) > 0 {
+			return nil, fmt.Errorf("you may not set --node-name when debugging a node")
+		}
+		if o.AsNonRoot || o.AsUser > 0 {
+			// TODO: allow --as-root=false to skip all the namespaces except network
+			return nil, fmt.Errorf("can't debug nodes without running as the root user")
+		}
+		image := "registry.access.redhat.com/rhel7/rhel-tools"
+		// container image is overwritten, but avoid
+		if len(o.Image) == 0 {
+			istag, err := o.ImageClient.ImageStreamTags("openshift").Get("tools:latest", metav1.GetOptions{})
+			if err == nil && len(istag.Image.DockerImageReference) > 0 {
+				image = istag.Image.DockerImageReference
+			}
+		}
+		zero := int64(0)
+		isTrue := true
+		hostPathType := corev1.HostPathDirectory
+		return &corev1.PodTemplateSpec{
+			Spec: corev1.PodSpec{
+				NodeName:    t.Name,
+				HostNetwork: true,
+				HostPID:     true,
+				Volumes: []corev1.Volume{
+					{
+						Name: "host",
+						VolumeSource: corev1.VolumeSource{
+							HostPath: &corev1.HostPathVolumeSource{
+								Path: "/",
+								Type: &hostPathType,
+							},
+						},
+					},
+				},
+				RestartPolicy: corev1.RestartPolicyNever,
+				Containers: []corev1.Container{
+					{
+						Name:  "container-00",
+						Image: image,
+						Resources: corev1.ResourceRequirements{
+							Requests: corev1.ResourceList{
```

Comment:
Do we need to set these?

Reply:
We need to account for the resources we actually use, but without adding `--resources` to allow an admin to override it's probably too early.  The tradeoff is that we might get evicted if the node is under pressure, but it's slightly better for the workloads on the node if the admin debug pod is what gets evicted, not the workload.  

We will have to have some future way to ensure we don't get evicted, but I don't want to design it yet until we know how frequent this issue is.  As a workaround an admin can always `-o yaml` and edit the pod, then start and attach.